src/dmangle.d

```diff
@@ -88,6 +94,9 @@ private immutable char[TMAX] mangleChar =
     //              X   // variadic T t...)
     //              Y   // variadic T t,...)
     //              Z   // not variadic, end of parameters
+    //              #   // Type backward reference
```

Comment:
> ambiguities ?

A simple bugzilla search already lists 4: https://issues.dlang.org/buglist.cgi?quicksearch=mangl%20ambig

> Do they pose a problem ?

Sure. Not sure whether it gets worse with back references.